### PR TITLE
feat: ChangeTierRequest updates MasterUserRecord.Spec.TierName

### PR DESCRIPTION
**Pull request by xcoulon**
Fixes https://issues.redhat.com/browse/CRT-1322

See also https://github.com/codeready-toolchain/toolchain-common/pull/218

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
No E2E tests to add, since the changes in the migration of MasterUserRecords done by the UserSignupController were tested (see https://github.com/codeready-toolchain/toolchain-e2e/pull/411). In other words, the assertions added in https://github.com/codeready-toolchain/toolchain-e2e/pull/411 still work, but this time because of the ChangeTierRequestController sets the `MasterUserRecord.Spec.TierName` field directly.

File: controllers/changetierrequest/changetierrequest_controller.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func (r *Reconciler) changeTier(logger logr.Logger, changeTierRequest *toolchain
 				// here we remove the template hash label since it was change for one or all target clusters
 				delete(mur.Labels, nstemplatetier.TemplateTierHashLabelKey(mur.Spec.UserAccounts[i].Spec.NSTemplateSet.TierName))
 				mur.Spec.UserAccounts[i].Spec.NSTemplateSet = newNsTemplateSet
+				mur.Spec.TierName = changeTierRequest.Spec.TierName
```
Comment (MatousJobanek) [inline controllers/changetierrequest/changetierrequest_controller.go:171]: why change it in the for-loop that goes over all UserAccount? wouldn't make more sense to change it outside of the for-loop? The particular `mur.Spec.TierName` field shouldn't be related to UserAccount anymore.

File: controllers/changetierrequest/changetierrequest_controller.go [hunk 2]
```diff
@@ -176,6 +177,7 @@ func (r *Reconciler) changeTier(logger logr.Logger, changeTierRequest *toolchain
 			// here we remove the template hash label since it was change for one or all target clusters
 			delete(mur.Labels, nstemplatetier.TemplateTierHashLabelKey(mur.Spec.UserAccounts[i].Spec.NSTemplateSet.TierName))
 			mur.Spec.UserAccounts[i].Spec.NSTemplateSet = newNsTemplateSet
+			mur.Spec.TierName = changeTierRequest.Spec.TierName
```
Comment (MatousJobanek) [inline controllers/changetierrequest/changetierrequest_controller.go:180]: it's the same. 
Actually, both of them should be replaced by one assignment outside of the for-loop.

File: controllers/changetierrequest/changetierrequest_controller_test.go [hunk 3]
```diff
@@ -53,6 +53,7 @@ func TestChangeTierSuccess(t *testing.T) {
 		// then
 		require.NoError(t, err)
 		murtest.AssertThatMasterUserRecord(t, "john", cl).
+			HasTier(*teamTier).
```
Comment (MatousJobanek) [inline controllers/changetierrequest/changetierrequest_controller_test.go:56]: it's just a detail, but we don't have to pass the whole `NSTemplateTier` instance - just the name is enough

**Review by MatousJobanek — COMMENTED** [review]
Sorry for my late review

File: controllers/changetierrequest/changetierrequest_controller_test.go (hunk 3 above)
Comment (xcoulon) [reply controllers/changetierrequest/changetierrequest_controller_test.go:56]: the `HasTier` func is in toolchain-common. We can change that later if it causes troubles.

File: controllers/changetierrequest/changetierrequest_controller.go (hunk 2 above)
Comment (xcoulon) [reply controllers/changetierrequest/changetierrequest_controller.go]: done in 9847f5d

File: controllers/changetierrequest/changetierrequest_controller.go [hunk 4]
```diff
@@ -178,6 +178,7 @@ func (r *Reconciler) changeTier(logger logr.Logger, changeTierRequest *toolchain
 			mur.Spec.UserAccounts[i].Spec.NSTemplateSet = newNsTemplateSet
 		}
 	}
+	mur.Spec.TierName = changeTierRequest.Spec.TierName
```
Comment (MatousJobanek) [inline controllers/changetierrequest/changetierrequest_controller.go:181]: shouldn't this have been:
```suggestion
	if mur.Spec.TierName != changeTierRequest.Spec.TierName {
		mur.Spec.TierName = changeTierRequest.Spec.TierName
		changed = true
	}
```

**Review by MatousJobanek — APPROVED** [review]
:+1: :rocket:

**Pull request merged**